pdata/pprofile/attributes.go

```diff
@@ -26,3 +30,35 @@ func FromAttributeIndices(table AttributeTableSlice, record attributable) pcommo
 
 	return m
 }
+
+// AddAttribute updates an AttributeTable and a record's AttributeIndices to
+// add a new attribute.
+// The record can by any struct that implements an `AttributeIndices` method.
+func AddAttribute(table AttributeTableSlice, record attributable, key string, value any) error {
+	for i := 0; i < table.Len(); i++ {
```

Comment:
```suggestion
	for i := range table.Len() {
```

Reply:
Every other loop through slices in pdata uses the old for loop. I'd rather not deviate from the convention.

Reply:
Take it as a nit, not a blocker.
Is there really a "convention" or is it more a copy&paste thing everywhere? 🤷🏻